dvc/remote/local/__init__.py

```diff
@@ -56,10 +58,18 @@ class RemoteLOCAL(RemoteBASE):
         "reflink": System.reflink,
     }
 
+    SHARED_MODE_MAP = {None: (0o644, 0o755), "group": (0o664, 0o775)}
+
     def __init__(self, repo, config):
         super(RemoteLOCAL, self).__init__(repo, config)
         self.protected = config.get(Config.SECTION_CACHE_PROTECTED, False)
 
+        shared = config.get(Config.SECTION_CACHE_SHARED)
+        self._file_mode, self._dir_mode = self.SHARED_MODE_MAP[shared]
+
+        if self.protected:
+            self._file_mode = stat.S_IREAD | stat.S_IRGRP | stat.S_IROTH
```

Comment:
maybe adding a comment here explaining a little bit the _bitfield_ and what is the intention behind those flags